src/lang/base/Syntax.ml

```diff
@@ -202,6 +202,138 @@ type literal =
                CPSMonad.t)
 [@@deriving sexp]
 
+(* Builtins *)
+type builtin =
+  | Builtin_eq
+  | Builtin_concat
+  | Builtin_substr
+  | Builtin_strlen
+  | Builtin_to_string
+  | Builtin_blt
+  | Builtin_badd
+  | Builtin_bsub
+  | Builtin_to_uint256
+  | Builtin_sha256hash
+  | Builtin_keccak256hash
+  | Builtin_ripemd160hash
+  | Builtin_to_bystr
+  | Builtin_schnorr_verify
+  | Builtin_ec_gen_key_pair (* in gas coster only *)
+  | Builtin_schnorr_gen_key_pair (* in cashflow checker only *)
+  | Builtin_schnorr_sign (* in cashflow checker only *)
+  | Builtin_ecdsa_verify
+  | Builtin_ecdsa_sign (* in gas coster only *)
+  | Builtin_contains
+  | Builtin_put
+  | Builtin_get
+  | Builtin_remove
+  | Builtin_to_list
+  | Builtin_size
+  | Builtin_lt
+  | Builtin_add
+  | Builtin_sub
+  | Builtin_mul
+  | Builtin_div
+  | Builtin_rem
+  | Builtin_pow
+  | Builtin_to_int32
+  | Builtin_to_int64
+  | Builtin_to_int128
+  | Builtin_to_int256
+  | Builtin_to_uint32
+  | Builtin_to_uint64
+  | Builtin_to_uint128
+  | Builtin_to_nat
+[@@deriving sexp]
+
+type 'rep tagged_builtin =
+  | TaggedBuiltin of builtin * 'rep
```

Comment:
This type should be called `builtin_annot` for consistency with the other annotated AST node types.

Also, if you move the type definition into the `ScillaSyntax` functor (defined slightly further down), then you can use the definition

```
type builtin_annot = builtin * ER.rep
```

instead. That way you don't need the `TaggedBuiltin` constructor.

Reply:
Neat. Thanks!!